stan/math/prim/prob/wiener5_lpdf.hpp

```diff
@@ -0,0 +1,428 @@
+ */
+template <size_t NestedIndex>
+inline void assign_err(double arg, double err) {
+  arg = err;
+}
+
+/**
+ * Utility function for replacing a value with a specified error value,
+ * overload for use when the value is stored within a tuple.
+ */
+template <size_t NestedIndex, typename... TArgs>
+inline void assign_err(std::tuple<TArgs...>& args_tuple, double err) {
+  std::get<NestedIndex>(args_tuple) = err;
+}
+
+/**
+ * Utility function for estimating a function with a given set of arguments,
+ * checking the result against a provided error tolerance, and re-estimating
+ * the function with the increased error if it fails.
+ *
+ * @tparam ErrIndex Position of the error argument in the provided arguments
+ * @tparam NestedIndex Nested position if the error argument is within a tuple
+ * @tparam F Type of functor
+ * @tparam ArgsTupleT Type of tuple of arguments for functor
+ *
+ * @param functor Function to apply
+ * @param err Error value to check against
+ * @param args_tuple Tuple of arguments to pass to functor
+ * @param log_result Whether the function result is already on the log-scale
+ */
+template <size_t ErrIndex, size_t NestedIndex = 0, typename F,
+          typename ArgsTupleT>
+double estimate_with_err_check(const F& functor, double err,
+                               ArgsTupleT&& args_tuple,
+                               bool log_result = true) {
+  double result = math::apply([&](auto&&... args) { return functor(args...); },
+                              args_tuple);
+  double lfabs_result = log_result ? log(fabs(result)) : fabs(result);
+  if (lfabs_result < err) {
+    lfabs_result = std::isinf(lfabs_result) ? 0 : lfabs_result;
+    ArgsTupleT err_args_tuple = args_tuple;
+    assign_err<NestedIndex>(std::get<ErrIndex>(err_args_tuple),
+                            err + lfabs_result);
+    result = math::apply([&](auto&&... args) { return functor(args...); },
+                         err_args_tuple);
+  }
+  return result;
+}
+}  // namespace internal
+
+/**
+ * Log-density function for the 5-parameter Wiener density.
+ * See 'wiener_full_lpdf' for more comprehensive documentation
+ *
+ * @tparam T_y type of scalar
+ * @tparam T_a type of boundary
+ * @tparam T_t0 type of non-decision time
+ * @tparam T_w type of relative starting point
+ * @tparam T_v type of drift rate
+ * @tparam T_sv type of inter-trial variability of drift rate
+ *
+ * @param y A scalar variable; the reaction time in seconds
+ * @param a The boundary separation
+ * @param t0 The non-decision time
+ * @param w The relative starting point
+ * @param v The drift rate
+ * @param sv The inter-trial variability of the drift rate
+ * @return The log of the Wiener first passage time density with
+ *  the specified arguments for upper boundary responses
+ */
+template <bool propto = false, typename T_y, typename T_a, typename T_t0,
+          typename T_w, typename T_v, typename T_sv>
+inline return_type_t<T_y, T_a, T_t0, T_w, T_v, T_sv> wiener5_lpdf(
```

Comment:
Is there a better name than `wiener5_lpdf`?

Reply:
This is the log density of the five parameter wiener diffusion model. The 5 refers to 5 parameters, there will also be wiener4 and wiener_full functions. (wiener4 probably for the cdf functions).